pandas/tests/groupby/test_grouping.py

```diff
@@ -442,6 +442,29 @@ def test_groupby_grouper_f_sanity_checked(self):
         with pytest.raises(AssertionError, match=msg):
             ts.groupby(lambda key: key[0:6])
 
+    def test_groupby_with_datetime_key(self):
+        # GH 51158
+        df = DataFrame(
+            {
+                "id": ["a", "b"] * 3,
+                "b": date_range("2000-01-01", "2000-01-03", freq="9H"),
+            }
+        )
+        grouper = Grouper(key="b", freq="D")
+        g = df.groupby([grouper, "id"])
```

Comment:
nitpick: can you call this "gb" instead of "g"?  1-letter variable names are tough to grep for